mlflow/store/sqlalchemy_store.py

```diff
@@ -463,6 +463,19 @@ def set_tag(self, run_id, tag):
             self._check_run_is_active(run)
             session.merge(SqlTag(run_uuid=run_id, key=tag.key, value=tag.value))
 
+    def delete_tag(self, run_id, key):
+        with self.ManagedSessionMaker() as session:
+            run = self._get_run(run_uuid=run_id, session=session)
+            self._check_run_is_active(run)
+            filtered_tags = session.query(SqlTag).filter_by(key=key).all()
+            if len(filtered_tags) == 0:
+                raise MlflowException(
+                    "No tag with name: {} in run with id {}".format(key, run_id))
```

Comment:
This exception should have `error_code = mlflow.protos.databricks_pb2.RESOURCE_DOES_NOT_EXIST`